File: Data Structures & Algorithms/combination-target-sum-ii/submission-4.java
class Solution {
    private List<List<Integer>> res ;
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        res=new ArrayList<>();
        Arrays.sort(candidates);
        helper(candidates,target,0,0, new ArrayList<Integer>());

        return res;

        
    }
    private void helper(int[] candidates, int target, int total,int start,List<Integer> curr) {
        if(total==target){
            res.add(new ArrayList<>(curr));
            return ;
        }
        
        for (int i =start ;i<candidates.length;i++){
            if (candidates[i]+total>target){
                return ;
            }
            
            if (i > start && candidates[i] == candidates[i - 1]) {
                continue;
            }
            
            curr.add(candidates[i]);
            helper(candidates,target,total+candidates[i],i+1,curr);
            curr.remove(curr.size()-1);
        }


    }
}

File: Data Structures & Algorithms/assign-cookies/submission-4.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        s.sort()
        g.sort()
        i=0
        j=0
        res=0
        while i<len(g):
            while j<len(s) and s[j]<g[i]:
                j+=1
            if j==len(s):
                return res
            res+=1
            i+=1
        return res
        